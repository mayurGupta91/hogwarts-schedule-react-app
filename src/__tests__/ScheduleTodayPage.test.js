/* eslint-env jest */

// ScheduleTodayPage.test.js

import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { shallow } from 'enzyme';
import ScheduleTodayPage from '../App';

describe('ScheduleTodayPage Component', () => {
    test('renders ScheduleTodayPage component snapshot', () => {
        const component = <ScheduleTodayPage />;
        expect(shallow(component)).toMatchSnapshot();
    });

    test('handles attendance change for each teacher', () => {
        const { getAllByDisplayValue } = render(<ScheduleTodayPage />);

        const presentElements = getAllByDisplayValue('Present');
        fireEvent.change(presentElements[5], { target: { value: 'Absent' } });
        fireEvent.change(presentElements[4], { target: { value: 'Absent' } });
        fireEvent.change(presentElements[3], { target: { value: 'Absent' } });
        fireEvent.change(presentElements[2], { target: { value: 'Absent' } });
        fireEvent.change(presentElements[1], { target: { value: 'Absent' } });
        fireEvent.change(presentElements[0], { target: { value: 'Absent' } });

        const absentElements = getAllByDisplayValue('Absent');
        fireEvent.change(absentElements[5], { target: { value: 'Present' } });
        fireEvent.change(absentElements[4], { target: { value: 'Present' } });
        fireEvent.change(absentElements[3], { target: { value: 'Present' } });
        fireEvent.change(absentElements[2], { target: { value: 'Present' } });
        fireEvent.change(absentElements[1], { target: { value: 'Present' } });
        fireEvent.change(absentElements[0], { target: { value: 'Present' } });
    });
});
