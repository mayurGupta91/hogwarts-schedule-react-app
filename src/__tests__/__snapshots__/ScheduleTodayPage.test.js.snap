// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleTodayPage Component renders ScheduleTodayPage component snapshot 1`] = `ShallowWrapper {}`;
